openmc/data/neutron.py

```diff
@@ -0,0 +1,926 @@
+
+        # Read fission energy release data
+        if 'fission_energy_release' in group:
+            fer_group = group['fission_energy_release']
+            data.fission_energy = FissionEnergyRelease.from_hdf5(fer_group)
+
+        return data
+
+    @classmethod
+    def from_ace(cls, ace_or_filename, metastable_scheme='nndc'):
+        """Generate incident neutron continuous-energy data from an ACE table
+
+        Parameters
+        ----------
+        ace_or_filename : openmc.data.ace.Table or str
+            ACE table to read from. If the value is a string, it is assumed to
+            be the filename for the ACE file.
+        metastable_scheme : {'nndc', 'mcnp'}
+            Determine how ZAID identifiers are to be interpreted in the case of
+            a metastable nuclide. Because the normal ZAID (=1000*Z + A) does not
+            encode metastable information, different conventions are used among
+            different libraries. In MCNP libraries, the convention is to add 400
+            for a metastable nuclide except for Am242m, for which 95242 is
+            metastable and 95642 (or 1095242 in newer libraries) is the ground
+            state. For NNDC libraries, ZAID is given as 1000*Z + A + 100*m.
+
+        Returns
+        -------
+        openmc.data.IncidentNeutron
+            Incident neutron continuous-energy data
+
+        """
+
+        # First obtain the data for the first provided ACE table/file
+        if isinstance(ace_or_filename, Table):
+            ace = ace_or_filename
+        else:
+            ace = get_table(ace_or_filename)
+
+        # If mass number hasn't been specified, make an educated guess
+        zaid, xs = ace.name.split('.')
+        if not xs.endswith('c'):
+            raise TypeError(
+                "{} is not a continuous-energy neutron ACE table.".format(ace))
+        name, element, Z, mass_number, metastable = \
+            get_metadata(int(zaid), metastable_scheme)
+
+        # Assign temperature to the running list
+        kTs = [ace.temperature*EV_PER_MEV]
+
+        data = cls(name, Z, mass_number, metastable,
+                   ace.atomic_weight_ratio, kTs)
+
+        # Get string of temperature to use as a dictionary key
+        strT = data.temperatures[0]
+
+        # Read energy grid
+        n_energy = ace.nxs[3]
+        i = ace.jxs[1]
+        energy = ace.xss[i : i + n_energy]*EV_PER_MEV
+        data.energy[strT] = energy
+        total_xs = ace.xss[i + n_energy : i + 2*n_energy]
+        absorption_xs = ace.xss[i + 2*n_energy : i + 3*n_energy]
+        heating_number = ace.xss[i + 4*n_energy : i + 5*n_energy]*EV_PER_MEV
+
+        # Create redundant reactions (total, absorption, and heating)
+        total = Reaction(1)
+        total.xs[strT] = Tabulated1D(energy, total_xs)
+        total.redundant = True
+        data.reactions[1] = total
+
+        if np.count_nonzero(absorption_xs) > 0:
+            absorption = Reaction(101)
```

Comment:
Optional: I would clarify the significance of `101` and `301` in this circumstance for readability.